### ISSUE TITLE
Expose class names and special strings

### ISSUE THREAD

**Issue by joshualamusga1** [microsoft/fluentui#10444]
#### Describe the feature that you would like added
It would improve maintainability to expose class names like ms-List as variables. The class names could then be changed more easily, and when variables are removed in a breaking change, it's much more clear to the consuming app, which will be able to remove the reference. Special strings should also be publicly exported when they're expected to be used, such as the constants in SelectionZone.base.tsx, e.g. "data-selection-disabled".

#### What component or utility would this be added to
Anywhere a unique class name or string is made up.

#### Have you discussed this feature with our team, and if so, who
Not previously discussed.

**Comment by natalieethell** [comment]
This seems like a valid ask to me - @dzearing @micahgodbolt could you take a look at this one? Has exposing these string constants been discussed?

**Comment by dzearing** [comment]
This is totally valid feedback; if there are spot fixes we can make to unblock a scenario, we're happy to take PRs or fix things for you.

But I'd like to get your take on the approach here. Exposing the classnames as an export means that we plan on using them forever. It's hard to make everyone happy. We ran into numerous customer issues just having the global classnames, and some asked for them to be removed altogether. The reason being the repeating problem with global css: works great for a single small app, but with many team checking in code to a single app, all it takes is one dev to target ms-Button with some min-width that looks great in their scenario to break everyone else.

A counter proposal is: bring your own through a common property like "classes".

```tsx
<Button classes={{ root: 'foo', rootPrimary: 'foo-primary', icon: 'bar', }} />
```

So then, Fabric doesn't define the classnames for you; you provide your own css classes, which can be injected any way you'd like.

**Comment by joshualamusga1** [comment]
You're right. If there is a bug or issue that involves changing a class name, making the promise that they belong to the public interface would turn that into a major version change.

The counter-proposal doesn't cover much ground that defining `styles` or using a React ref can't accomplish. I raised this idea because I had to write selectors like `.ms-DetailsRow &:hover`, where I don't directly render the details row component myself, and there is no exposed class name. Anything that helps make scenarios like that possible would be fine.
